@xen-orchestra/lite/src/stores/xen-api/network.store.ts

```diff
@@ -1,9 +1,105 @@
 import { defineStore } from 'pinia'
+import { computed } from 'vue'
+import { useI18n } from 'vue-i18n'
 
 export const useNetworkStore = defineStore('xen-api-network', () => {
-  const config = createXapiStoreConfig('network')
+  const { t } = useI18n()
 
-  return createSubscribableStoreContext(config, {})
+  const deps = {
+    poolStore: usePoolStore(),
+    hostStore: useHostStore(),
+    pifStore: usePifStore(),
+  }
+
+  const { context: baseContext, ...configRest } = createXapiStoreConfig('network', {
+    sortBy: sortByNameLabel,
+  })
+
+  const poolContext = deps.poolStore.getContext()
+  const hostContext = deps.hostStore.getContext()
+  const pifContext = deps.pifStore.getContext()
+
+  const PIFsByNetwork = computed(() => {
+    const PIFsByNetworkMap = new Map<string, XenApiPif[]>()
+
+    const poolMasterRef = poolContext.pool.value?.master
+
+    const currentHostRef = hostContext.records.value.find(host => host.$ref === poolMasterRef)
+    if (currentHostRef) {
+      pifContext.records.value
+        .filter(pif => pif.host === poolMasterRef)
+        .forEach(pif => {
+          if (!PIFsByNetworkMap.has(pif.network)) {
+            PIFsByNetworkMap.set(pif.network, [])
+          }
+          PIFsByNetworkMap.get(pif.network)?.push(pif)
+        })
+    }
+    return PIFsByNetworkMap
+  })
+
+  const networksWithVLANs = computed(() => {
+    const networksInfoMap = new Map<
+      string,
+      {
+        network: XenApiNetwork
+        vlan: string
+        status: Status
+      }
+    >()
+
+    return baseContext.records.value
+      .filter(network => network.PIFs.length > 0)
+      .map(network => {
+        const relatedPifs = PIFsByNetwork.value.get(network.$ref) || []
+        const vlan =
+          relatedPifs.length > 0 ? (relatedPifs[0].VLAN === -1 ? t('none') : relatedPifs[0].VLAN.toString()) : ''
+        const status = determineStatus(relatedPifs)
+
+        const networkWithDetails = {
+          network,
+          vlan,
+          status,
+        }
+        if (!networksInfoMap.has(network.$ref)) {
+          networksInfoMap.set(network.$ref, networkWithDetails)
+        }
+        networksInfoMap.set(network.$ref, networkWithDetails)
+        return networkWithDetails
+      })
+  })
+
+  const hostInternalNetworks = computed(() => {
+    return baseContext.records.value.filter(network => network.PIFs.length === 0) // Only networks without PIFs
+  })
+
+  const context = {
+    ...baseContext,
+    networksWithVLANs,
+    hostInternalNetworks,
+  }
+
+  return createSubscribableStoreContext({ context, ...configRest }, deps)
 })
+
+function determineStatus(PIFs: XenApiPif[]): Status {
+  if (PIFs.length === 0) {
+    return 'disconnected'
+  }
+  const currentlyAttached = PIFs.map(PIF => PIF.currently_attached)
+  if (currentlyAttached.every(Boolean)) {
+    return 'connected'
+  }
+  if (currentlyAttached.some(Boolean)) {
+    return 'partial'
+  }
+  return 'disconnected'
+}
```

Comment:
Since you have a `pif` store, why did you put this in the network store?

```suggestion
function determineStatus(PIFs: XenApiPif[]): Status {
  if (PIFs.length === 0) {
    return 'disconnected'
  }
  const currentlyAttached = PIFs.map(PIF => PIF.currently_attached)
  if (currentlyAttached.every(Boolean)) {
    return 'connected'
  }
  if (currentlyAttached.some(Boolean)) {
    return 'partial'
  }
  return 'disconnected'
}
```